src/com/mambu/apisdk/util/RequestExecutorImpl.java

```diff
@@ -139,21 +141,22 @@ public String executeRequest(String urlString, ParamsMap params, Method method,
 		} catch (IOException e) {
 			LOGGER.warning("IOException: message= " + e.getMessage());
 			throw new MambuApiException(e);
+		} finally {
+			httpClient.getConnectionManager().shutdown();
```

Comment:
Should be. This should be done as a separate refactoring effort to do regression testing and to test for multithreading. potential leaks, etc.